js/src/components/enable-new-product-sync-button.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import { glaData } from '.~/constants';
+import { API_NAMESPACE } from '.~/data/constants';
+import useApiFetchCallback from '.~/hooks/useApiFetchCallback';
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+
+/**
+ * Button to initiate auth process for WP Rest API
+ *
+ * @param {Object} params The component params
+ * @return {JSX.Element} The button.
+ */
+const EnableNewProductSyncButton = ( params ) => {
+	const { createNotice } = useDispatchCoreNotices();
+
+	const nextPageName = glaData.mcSetupComplete ? 'settings' : 'setup-mc';
+	const query = { next_page_name: nextPageName };
+	const path = addQueryArgs( `${ API_NAMESPACE }/rest-api/authorize`, query );
+	const [ fetchRestAPIAuthorize ] = useApiFetchCallback( { path } );
+	const handleEnableClick = async () => {
+		try {
+			const d = await fetchRestAPIAuthorize();
+			window.location.href = d.auth_url;
```

Comment:
Similar to https://github.com/woocommerce/google-listings-and-ads/pull/2311/files#r1539571082

Not sure if we should use the hook here

Reply:
Same as above, the code here is getting the auth status by calling GET auth status API so I think the hook `useUpdateRestAPIAuthorizeStatusByUrlQuery` couldn't be used here.